bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/connection/ConnectionManager.java

```diff
@@ -0,0 +1,530 @@
+                                    responseType = "exception";
+                                    break;
+                                case 0x07:
+                                    responseType = "querySN";
+                                    break;
+                                case 0x0A:
+                                    responseType = "exception2";
+                                    break;
+                                case 0x09: // Helyesen: 0xA0
+                                    responseType = "querySubtype";
+                                    break;
+                                default:
+                                    logger.debug("Invalid response type: {}", data[0x9]);
+                            }
+                            logger.trace("Response Type: {} and bodyType: {}", responseType, bodyType2);
+
+                            // The response data from the appliance includes a packet header which we don't want
+                            data = Arrays.copyOfRange(data, 10, data.length);
+                            byte bodyType = data[0x0];
+                            logger.trace("Response Type expected: {} and bodyType: {}", responseType, bodyType);
+                            logger.trace("Bytes in HEX, decoded and stripped without header: length: {}, data: {}",
+                                    data.length, Utils.bytesToHex(data));
+                            logger.debug("Bytes in BINARY, decoded and stripped without header: length: {}, data: {}",
+                                    data.length, Utils.bytesToBinary(data));
+
+                            if (data.length < 21) {
+                                logger.warn("Response data is {} long, minimum is 21!", data.length);
+                                return;
+                            }
+                            if (bodyType != -64) {
+                                if (bodyType == 30) {
+                                    logger.warn("Error response 0x1E received {} from IP Address:{}", bodyType,
+                                            ipAddress);
+                                    return;
+                                }
+                                logger.warn("Unexpected response bodyType {}", bodyType);
+                                return;
+                            }
+                            lastResponse = new Response(data, version, responseType, bodyType);
+                            try {
+                                logger.trace("Data length is {}, version is {}, IP address is {}", data.length, version,
+                                        ipAddress);
+                                if (callback != null) {
+                                    callback.updateChannels(lastResponse);
+                                }
+                            } catch (Exception ex) {
+                                logger.warn("Processing response exception: {}", ex.getMessage());
```

Comment:
What kind of exception do we expect here? This catches all kind of exceptions. I guess we should throw a `MideaException` here?

Reply:
I got an error message when with that 
![image](https://github.com/user-attachments/assets/99bd79ec-f0ef-4752-a9bc-57c90b255df7)

Reply:
Throw not catch

Reply:
Think I have that fixed